torch/_dynamo/variables/optimizer.py

```diff
@@ -166,8 +166,13 @@ def map_sources_and_install_guards(self, tx):
         # Mark all the tensors in the state dict to be static address. This has
         # to be done first because the variable builder relies on the static
         # address annotation.
+        # Note: we don't mark complex params as static addresses
+        # The main reason for this is that both a real view
+        # and the original complex tensor will share their data_ptr
+        # so we should guard on the dtype in this case.
         def mark_static(x):
-            mark_static_address(x, guard=False)
+            if not x.is_complex():
+                mark_static_address(x)
 
         tree_map_only(torch.Tensor, mark_static, self.value.state)
```

Comment:
But we do this here already, @mlazos?

Reply:
Maybe move this to init as well?

Reply:
Discussed offline, treemap doesn't iterate over keys :(